component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -151,13 +152,16 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             throws AuthenticationFailedException {
 
         try {
+            PrivilegedCarbonContext.startTenantFlow();
```

Comment:
Do everything in this block require to go through the tenant flow? or else better to wrap only the email sending part within the carbon context `try-final` block.

Reply:
Normally we used  try-final blocks in all the places start and end the tenant flow